src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -158,21 +232,21 @@ private BoundExpression BindDeconstructionAssignment(AssignmentExpressionSyntax
         private class DeconstructionVariable
         {
             public readonly BoundExpression Single;
-            public readonly ImmutableArray<DeconstructionVariable> Nested;
+            public readonly ArrayBuilder<DeconstructionVariable> Nested;
 
             public DeconstructionVariable(BoundExpression variable)
             {
                 Single = variable;
-                Nested = default(ImmutableArray<DeconstructionVariable>);
+                Nested = null;
             }
 
-            public DeconstructionVariable(ImmutableArray<DeconstructionVariable> variables)
+            public DeconstructionVariable(ArrayBuilder<DeconstructionVariable> variables)
             {
                 Single = null;
                 Nested = variables;
             }
 
-            public bool IsNested => !Nested.IsDefault;
+            public bool IsNested => Nested != null;
```

Comment:
`HasNestedVariables` sounds good to me.

---
In reply to: [68803104](https://github.com/dotnet/roslyn/pull/12224#discussion_r68803104) [](ancestors = 68803104)